plugins/inventory/vmware_vm_inventory.py

```diff
@@ -343,7 +343,7 @@ def __init__(self, hostname, username, password, port, validate_certs, with_tags
         self.hostname = hostname
         self.username = username
         self.password = password
-        self.port = port
+        self.port = int(port)
```

Comment:
`port` is set using `self.get_option`, we can specify the data type of parameter in the documentation and `get_option` will take care of typecasting.

```diff
diff --git a/plugins/inventory/vmware_vm_inventory.py b/plugins/inventory/vmware_vm_inventory.py
index bd24435..d736f10 100644
--- a/plugins/inventory/vmware_vm_inventory.py
+++ b/plugins/inventory/vmware_vm_inventory.py
@@ -46,6 +46,7 @@ DOCUMENTATION = r'''
         port:
             description: Port number used to connect to vCenter or ESXi Server.
             default: 443
+            type: int
             env:
               - name: VMWARE_PORT
         validate_certs:
```

Can you try this once and let me know if it works for you? Add a changelog entry [here](https://github.com/ansible-collections/community.vmware/tree/main/changelogs/fragments)

Reply:
@Akasurde thanks for your update. Yes, casting it directly on the `get_option` did the trick. I've updated the patch and also included it on the changelog. Please let me know if looks now. Thank you!